source/agora/consensus/data/ValidatorBlockSig.d

```diff
@@ -0,0 +1,60 @@
+/*******************************************************************************
+
+    Contains definition for the `ValidatorBlockSig` struct,
+    which is used to communicate block signatures between nodes.
+
+    Copyright:
+        Copyright (c) 2019-2020 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.consensus.data.ValidatorBlockSig;
+
+import agora.common.Types;
+import agora.common.crypto.ECC;
+import agora.common.crypto.Key;
+import agora.common.crypto.Schnorr;
+
+/*******************************************************************************
+
+    Define Validator Block Signature information
+
+*******************************************************************************/
+
+public struct ValidatorBlockSig
```

Comment:
I'm guessing it's not possible to just use the block hash rather than the `height + challenge`? It could save a bit of bandwidth.

Reply:
Maybe, however I wanted to be very sure that the signatures for different potential blocks wouldn't get combined and then invalidate the block signature.

Reply:
Actualy dropped the scalar from the structure. If a validator signs a different block for the same height then it will not be included anyway as the signature will not be accepted. (Although I think I need to add that check in the relavent commit still)